src/components/Books.js

```diff
@@ -65,21 +77,46 @@ const removeBook = async (title) => {
 
 const addBooks = async ({title, authors, description, published, image, link}) => {
   try{
-    console.log("USER",user);
-      const book = {title, authors, description, published, image, link, userId:user.attributes.sub}
-      let response = await API.graphql(graphqlOperation(createBook, {input:book}))
-      fetchBooks()
-      toast.success(`${title} Added Successfully`);
-      console.log(response);
+    console.log("USER",userId);
+      const book = {title, authors, description, published, image, link}
+      let  allBooks = await API.graphql(graphqlOperation(listBooks,{
+          filter : {
+            title : {
+              eq: title
```

Comment:
Maybe you can try filtering by ISBN instead of title. That would give us more uniqueness.

Reply:
ISBN is nested three query deep and as many forms of it so I used etag for simplicity which is unique for every book